### PR TITLE
refactor(webserver): Replace rust-crypto crate

### LINKED ISSUE #165: Stop using rust-crypto
As far as I know, `rust-crypto` is only used for the SHA1 HMAC for rewriting image URLs with camo. `rust-crypto` is unmaintained and old. Using a combination of the [`hmac`][hmac] crate and the [`sha-1`][sha1] crate is recommended now.

Anywhere there's `use crypto::*` should be updated.

Edit: MD5 is used somewhere for some reason, too (probably gravatar). [`md-5`][md5] is recommended.

[hmac]: https://docs.rs/hmac/0.7.1/hmac/
[sha1]: https://docs.rs/sha-1/0.8.1/sha1/
[md5]: https://docs.rs/md-5/0.8.0/md5/

### PR DESCRIPTION
**Pull request by kitlith**
Use sha-1, hmac, and md-5 from the RustCrypto project instead.

Closes #165 

For the credits, is it okay to link to the RustCrypto general pages instead of, say, the docs pages? Should this be considered chore(deps) rather than refactor?

### CODE REVIEW

File: webserver/src/models/user.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-use crypto::{digest::Digest, md5::Md5};
+use md5::{Md5, Digest};
```
Comment (jkcclemens) [inline webserver/src/models/user.rs]: This is super nitpicky, but I this import should be between `failure` and `rocket`.

I like to go alphabetical crates, local crate, std as the order.

It should also be in alphabetical order itself, as below:

```suggestion
use md5::{Digest, Md5};
```

**Comment by kitlith** [comment]
The question was more about the url being the github repo https://github.com/RustCrypto/hashes multiple times, instead of e.g. linking to the docs.rs page for each crate. I did the former since you seem to prefer linking to the github pages... *shrug*

Do I also need to change the order in Cargo.toml?

**Comment by jkcclemens** [comment]
No, Cargo.toml has no ordering at the moment. And I read this PR just after I woke up and was going to work, so excuse my misunderstanding of your question. Linking to the GitHub general pages is fine.

**Comment by kitlith** [comment]
Alright, build passed, I think it should be good in that case, then.

**Pull request merged**